src/result_processor.c

```diff
@@ -845,3 +846,203 @@ ResultProcessor *RPCounter_New() {
   ret->base.type = RP_COUNTER;
   return &ret->base;
 }
+
+/*******************************************************************************************************************
+ *  Buffer and Locker Results Processor
+ *
+ * This component should be added to the pipeline of the query excution, if a thread safe access to
+ * Redis keyspace is required.
+ *
+ * The buffer is responsible for buffering the document that pass the query filters and lock the GIL
+ * to allow the downstream result processor safe access to redis keyspace.
+ *
+ * Unlocking the GIL should be done only by the Unlocker result processor.
+ *******************************************************************************************************************/
+
+typedef struct {
+  ResultProcessor base;
+
+  FixedSizeBlocksManager bufferBlocks;
+  FixedSizeBlocksIterator resultsIterator;
+  size_t BlockSize;
+
+} RPBufferAndLocker;
+
+static int rpbufferNext_bufferDocs(ResultProcessor *rp, SearchResult *res);
+
+static void RPBufferAndLocker_Free(ResultProcessor *base) {
+  RPBufferAndLocker *this = (RPBufferAndLocker *)base;
+
+  // Free buffer memory blocks
+  FixedSizeBlocksManager_FreeAll(&this->bufferBlocks);
+
+  // Invalidate the iterator
+  FixedSizeBlocksManager_invalidateIterator(&this->resultsIterator);
```

Comment:
I think this line is redundant as this object is going to be deleted and ivalidate just nullifies members